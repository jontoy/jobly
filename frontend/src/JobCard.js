import React from 'react';
import { Link } from 'react-router-dom';
const CompanyCard = ({id, title, equity, salary, company_handle, state, apply, unapply}) => {

    const applyButton = () => {
        return <button className="btn btn-success ml-auto" onClick={() => apply(id)}>Apply</button>
    }
    const unApplyButton = () => {
        return <button className="btn btn-danger ml-auto" onClick={() => unapply(id)}>Remove Application</button>
    }
    return (
        <div className="CompanyCard card p-2 mb-1">
            <h5 className="card-title">{title}</h5>
            {company_handle ? <h6 className="card-subtitle mb-2 text-muted"><Link to={`/companies/${company_handle}`}>(@{company_handle})</Link></h6> : null}
            <p className="card-text">
                {salary ? <span>Salary: {salary}</span> : null}
                <br/>
                {equity ? <span>Equity: {equity}</span> : null}
            </p>
            {state === 'applied' ? unApplyButton() : applyButton()}
        </div>
    )
}

export default CompanyCard;

