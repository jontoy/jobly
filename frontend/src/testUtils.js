import React from "react";
import UserContext from "./userContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs:[]
};

const UserProvider = ({ children, currentUser = demoUser, appliedIds = [] }) => (
  <UserContext.Provider value={{currentUser, appliedIds}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
