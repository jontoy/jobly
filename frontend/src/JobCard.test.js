import React from 'react';
import { render } from '@testing-library/react';
import JobCard from './JobCard';
import {UserProvider} from './testUtils';

it('renders without crashing', () => {
    render(
    <UserProvider>
        <JobCard/>
    </UserProvider>
    );
});
  
it('matches snapshot', () => {
      const {asFragment} = render(
      <UserProvider>
        <JobCard/>
      </UserProvider>);
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot when user has not applied to job', () => {
    const {asFragment} = render(
    <UserProvider>
        <JobCard id={1} title="fake job" salary={140000} equity={0.01}/>
    </UserProvider>
    );
  expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot when user has applied to job', () => {
    const {asFragment} = render(
    <UserProvider appliedIds={[1]}>
        <JobCard id={1} title="fake job" salary={140000} equity={0.01}/>
    </UserProvider>
    );
  expect(asFragment()).toMatchSnapshot();
});