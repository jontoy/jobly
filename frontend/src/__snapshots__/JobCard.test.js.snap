// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard card p-2 mb-1"
  >
    <h5
      class="card-title"
    />
    <p
      class="card-text"
    >
      <br />
    </p>
    <button
      class="btn btn-success ml-auto"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when user has applied to job 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard card p-2 mb-1"
  >
    <h5
      class="card-title"
    >
      fake job
    </h5>
    <p
      class="card-text"
    >
      <span>
        Salary: 140000
      </span>
      <br />
      <span>
        Equity: 0.01
      </span>
    </p>
    <button
      class="btn btn-danger ml-auto"
    >
      Remove Application
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when user has not applied to job 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard card p-2 mb-1"
  >
    <h5
      class="card-title"
    >
      fake job
    </h5>
    <p
      class="card-text"
    >
      <span>
        Salary: 140000
      </span>
      <br />
      <span>
        Equity: 0.01
      </span>
    </p>
    <button
      class="btn btn-success ml-auto"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;
